src/components/app/MenuButtons/Permalink.js

```diff
@@ -7,14 +7,19 @@
 import * as React from 'react';
 import ArrowPanel from '../../shared/ArrowPanel';
 import ButtonWithPanel from '../../shared/ButtonWithPanel';
-import { shortenUrl } from '../../../utils/shorten-url';
+import * as UrlUtils from '../../../utils/shorten-url';
+
+type Props = {|
+  +isNewlyPublished: boolean,
+  +injectedUrlShortener?: typeof UrlUtils.shortenUrl | void,
```

Comment:
I don't recall this approach in the codebase before. Can't we mock `shortenUrl` function during testing?